src/CommonLib/DirectoryObjects/DirectoryEntryWrapper.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.DirectoryServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Principal;
+using System.Text;
+
+namespace SharpHoundCommonLib;
+
+public class DirectoryEntryWrapper : IDirectoryObject {
+    private readonly DirectoryEntry _entry;
+
+    public DirectoryEntryWrapper(DirectoryEntry entry) {
+        _entry = entry;
+    }
+    
+    public bool TryGetDistinguishedName(out string value) {
+        return TryGetProperty(LDAPProperties.DistinguishedName, out value);
+    }
+
+    private bool CheckCache(string propertyName) {
+        try {
+            if (!_entry.Properties.Contains(propertyName))
+                _entry.RefreshCache(new[] { propertyName });
+
+            return _entry.Properties.Contains(propertyName);
+        }
+        catch {
+            return false;
+        }
+    }
+
+    public bool TryGetProperty(string propertyName, out string value) {
+        value = string.Empty;
+        if (!CheckCache(propertyName)) {
+            return false;
+        }
+
+        var s = _entry.Properties[propertyName][0];
```

Comment:
I see the other methods use `.Value` instead of `[0]`. I just wanted to verify this is correct?

Reply:
Good catch